src/main/java/dev/openfeature/sdk/NoOpProvider.java

```diff
@@ -10,6 +10,12 @@ public class NoOpProvider implements FeatureProvider {
     @Getter
     private final String name = "No-op Provider";
 
+    // The Noop provider is ALWAYS NOT_READY, otherwise READY handlers would run immediately when attached.
```

Comment:
No, the reason for the `NOT_READY` is so the SDK knows whether or not to initialize the provider and fire `READY` events, as dictated by the spec. It doesn't impact evaluation (the SDK still calls all the resolver methods).

I know this is a bit weird, but the Noop provider is really a special case - it only exists as a simple way to maintain the "no op" behavior when no "real" provider is set. An SDK could neglect to implement a no-op provider entirely and it would be compliant.

What this avoids is `READY` events firing immediately when no "real" provider has been set, which would be very confusing. I'd rather not have to add some hack to accomplish that.